docs/rules/0123/resource-pattern-singular.md

```diff
@@ -0,0 +1,85 @@
+---
+rule:
+  aip: 123
+  name: [core, '0123', resource-pattern-singular]
+  summary: Resource patterns must use the singular as the resource ID segment
+permalink: /123/resource-pattern-singular
+redirect_from:
+  - /0123/resource-pattern-singular
+---
+
+# Resource `pattern` use of `singular`
+
+This rule enforces that messages that have a `google.api.resource` annotation
+use the `singular` as the resource ID segment, as described in [AIP-123][].
```

Comment:
Add "form"? ("use the `singular` form as the resource ID segment")

Reply:
sure thing! Done